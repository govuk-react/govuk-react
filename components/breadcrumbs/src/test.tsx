import React from 'react';
import { mount } from 'enzyme';

import Breadcrumbs from '.';

describe('breadcrumbs', () => {
  it('should render an item in ordered list', () => {
    const wrapper = mount(<Breadcrumbs>example</Breadcrumbs>);

    expect(wrapper.find('ol')).toHaveLength(1);
  });

  it('should render multiple items in ordered list', () => {
    const wrapper = mount(
      <Breadcrumbs>
        <Breadcrumbs.Link href="/section">Section 1</Breadcrumbs.Link>
        example
      </Breadcrumbs>
    );

    expect(wrapper.find('ol li')).toHaveLength(2);
  });

  it('should render an ordered list without ghost/duff children', () => {
    const wrapper = mount(
      <Breadcrumbs>
        <Breadcrumbs.Link href="/section">Section 1</Breadcrumbs.Link>
        {[] /* empty node */}
        {null /* null node */}
        example
      </Breadcrumbs>
    );

    expect(wrapper.find('ol li')).toHaveLength(2);
  });

  it('allows attributes to be set', () => {
    const wrapper = mount(
      <Breadcrumbs id="test" className="test">
        Crumb
      </Breadcrumbs>
    );

    expect(wrapper.hasClass('test')).toBe(true);
    expect(wrapper.is('#test')).toBe(true);
  });

  it('matches snapshot', () => {
    const wrapper = mount(
      <Breadcrumbs>
        <Breadcrumbs.Link href="/section">Section 1</Breadcrumbs.Link>
        <Breadcrumbs.Link href="/section2">Section 2</Breadcrumbs.Link>
      </Breadcrumbs>
    );

    expect(wrapper).toMatchSnapshot('breadcrumbs');
  });
});
