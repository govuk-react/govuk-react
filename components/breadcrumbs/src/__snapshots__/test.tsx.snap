// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`breadcrumbs matches snapshot: breadcrumbs 1`] = `
.c3 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.c3:link,
.c3:visited,
.c3:hover,
.c3:active,
.c3:focus {
  color: #0b0c0c;
}

.c3:focus {
  outline: 3px solid #ffdd00;
  outline-offset: 0;
  background-color: #ffdd00;
}

.c0 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 14px;
  line-height: 1.1428571428571428;
  color: #0b0c0c;
  margin-top: 15px;
  margin-bottom: 10px;
}

.c1 {
  margin: 0;
  padding: 0;
  list-style-type: none;
  display: block;
}

.c2 {
  display: inline-block;
  position: relative;
  margin-bottom: 5px;
  margin-left: 10px;
  padding-left: 15.655000000000001px;
}

.c2::before {
  content: '';
  display: block;
  position: absolute;
  top: -1px;
  bottom: 1px;
  left: -3.3100000000000005px;
  width: 7px;
  height: 7px;
  margin: auto 0;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  border: solid;
  border-width: 1px 1px 0 0;
  border-color: #6f777b;
}

.c2:first-child {
  margin-left: 0;
  padding-left: 0;
}

.c2:first-child::before {
  content: none;
  display: none;
}

@media print {
  .c3 {
    font-family: sans-serif;
  }
}

@media print {
  .c3:link,
  .c3:visited,
  .c3:hover,
  .c3:active,
  .c3:focus {
    color: #000;
  }
}

@media print {
  .c0 {
    font-size: 14px;
    line-height: 1.2;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    font-size: 16px;
    line-height: 1.25;
  }
}

@media print {
  .c0 {
    color: #000;
  }
}

<Breadcrumbs>
  <src__BreadcrumbsContainer>
    <div
      className="c0"
    >
      <src__BreadcrumbsList>
        <ol
          className="c1"
        >
          <src__BreadcrumbsListItem
            key="0"
          >
            <li
              className="c2"
            >
              <link__Link
                href="/section"
              >
                <a
                  className="c3"
                  href="/section"
                >
                  Section 1
                </a>
              </link__Link>
            </li>
          </src__BreadcrumbsListItem>
          <src__BreadcrumbsListItem
            key="1"
          >
            <li
              className="c2"
            >
              <link__Link
                href="/section2"
              >
                <a
                  className="c3"
                  href="/section2"
                >
                  Section 2
                </a>
              </link__Link>
            </li>
          </src__BreadcrumbsListItem>
        </ol>
      </src__BreadcrumbsList>
    </div>
  </src__BreadcrumbsContainer>
</Breadcrumbs>
`;
